internal/saucecloud/cloud.go

```diff
@@ -934,3 +966,115 @@ func (r *CloudRunner) getHistory(launchOrder config.LaunchOrder) (insights.JobHi
 	}
 	return r.InsightsService.GetHistory(context.Background(), user, launchOrder)
 }
+
+func (r *CloudRunner) reportSuiteToInsights(res result) []insights.TestRun {
+	// Skip reporting if job is not completed
+	if !job.Done(res.job.Status) || res.skipped || res.job.ID == "" {
+		return []insights.TestRun{}
+	}
+
+	assets, err := r.JobService.GetJobAssetFileNames(context.Background(), res.job.ID, res.job.IsRDC)
+	if err != nil {
+		// TODO: Update message
+		log.Warn().Err(err).Msg(msg.InsightsReportError)
+	}
+
+	var testRuns []insights.TestRun
+
+	if arrayContains(assets, saucereport.SauceReportFileName) {
+		report, err := r.loadSauceTestReport(res.job.ID, res.job.IsRDC)
+		if err != nil {
+			log.Warn().Err(err).Str("action", "parsingJSON").Msg(msg.InsightsReportError)
+		}
+		testRuns = insights.FromSauceReport(report)
+	} else if arrayContains(assets, junit.JunitFileName) {
+		report, err := r.loadJUnitReport(res.job.ID, res.job.IsRDC)
+		if err != nil {
+			log.Warn().Err(err).Str("action", "parsingXML").Msg(msg.InsightsReportError)
+		}
+		testRuns = insights.FromJUnit(report)
+	}
+	enrichInsightTestRun(testRuns, res.job.ID, res.name, res.details, res.job.IsRDC)
+	if len(testRuns) > 0 {
+		if err := r.InsightsService.PostTestRun(context.Background(), testRuns); err != nil {
+			log.Warn().Err(err).Str("action", "posting").Msg(msg.InsightsReportError)
+		}
+	}
+	return testRuns
+}
+
+func (r *CloudRunner) loadSauceTestReport(jobID string, isRDC bool) (saucereport.SauceReport, error) {
+	fileContent, err := r.JobService.GetJobAssetFileContent(context.Background(), jobID, saucereport.SauceReportFileName, isRDC)
+	if err != nil {
+		log.Warn().Err(err).Str("action", "loading-json-report").Msg(msg.InsightsReportError)
+		return saucereport.SauceReport{}, err
+	}
+	return saucereport.Parse(fileContent)
+}
+
+func (r *CloudRunner) loadJUnitReport(jobID string, isRDC bool) (junit.TestSuites, error) {
+	fileContent, err := r.JobService.GetJobAssetFileContent(context.Background(), jobID, junit.JunitFileName, isRDC)
+	if err != nil {
+		log.Warn().Err(err).Str("action", "loading-xml-report").Msg(msg.InsightsReportError)
+		return junit.TestSuites{}, err
+	}
+	return junit.Parse(fileContent)
+}
+
+func arrayContains(list []string, want string) bool {
```

Comment:
I think you can use this method https://pkg.go.dev/golang.org/x/exp/slices#Contains

Reply:
This looks not stable yet, maybe we should avoid using it now ?